src/commands/DueDates/addDueDate.ts

```diff
@@ -0,0 +1,287 @@
+    if (typeof month.value === 'string') {
+      const tempDate = Date.parse(month.value + '1, 2021');
+      if (!isNaN(tempDate)) {
+        month = new Date(tempDate).getMonth() + 1;
+      } else return invalidSyntaxReply(message, settingsData);
+    }
+
+    // Set month variable to value for consistency
+    if (typeof month != 'number') {
+      month = month.value;
+    }
+
+    let date = new Date(year.value, (month as number) - 1, day.value);
+
+    // Sometimes an invalid date can be created but the date will change e.g Feb 29, 2021 becomes Mar 1, 2021. This doesn't let those cases through
+    if (!(date.getFullYear() === year.value) || !(date.getMonth() + 1 === month) || !(date.getDate() == day.value)) {
+      const embed = new BediEmbed()
+          .setColor(colors.ERROR)
+          .setTitle('Add Due Date Reply')
+          .setDescription('That date is invalid!');
+      return message.channel.send({embeds: [embed]});
+    }
+
+    const dateMoment = moment.tz(moment(date).format('YYYY-MM-DD'), settingsData.timezone);
+
+    let dateOnly = true;
+
+    if (timeString.success) {
+      if (!isValidTime(timeString.value)) {
+        const embed = new BediEmbed()
+            .setColor(colors.ERROR)
+            .setTitle('Add Due Date Reply')
+            .setDescription('That is not a valid time.');
+        return message.reply({embeds: [embed]});
+      }
+
+      const job = await agenda.schedule(timeString.value, 'Dummy Job', {});
+      const tempDate = job.attrs.nextRunAt;
+      await job.remove();
+
+      dateMoment.set({h: tempDate?.getHours(), m: tempDate?.getMinutes()});
+      dateOnly = false;
+    }
+
+    if (dateMoment < moment().subtract(1, 'd') || (!dateOnly && dateMoment < moment())) {
+      const embed = new BediEmbed()
+          .setColor(colors.ERROR)
+          .setTitle('Add Due Date Reply')
+          .setDescription('You can not set a date/time in the past.');
+      return message.reply({embeds: [embed]});
+    }
+
+    date = dateMoment.toDate();
+
+    const typeSelect = new MessageActionRow()
+        .addComponents(
+            new MessageSelectMenu()
+                .setCustomId('typeSelect')
+                .setPlaceholder('Select a Due Date Type')
+                .addOptions([
+                  {
+                    label: 'Assignment',
+                    value: 'Assignment',
+                  },
+                  {
+                    label: 'Test',
+                    value: 'Test',
+                  },
+                  {
+                    label: 'Quiz',
+                    value: 'Quiz',
+                  },
+                  {
+                    label: 'Exam',
+                    value: 'Exam',
+                  },
+                  {
+                    label: 'Project',
+                    value: 'Project',
+                  },
+                  {
+                    label: 'Other',
+                    value: 'Other',
```

Comment:
These should be user configurable. Some design teams and hackathons use Bedibot - this should be configurable as required